src/extensions/birthday_event.py

```diff
@@ -0,0 +1,292 @@
+        document = {
+            "guild_id": ctx.guild.id,
+            "member_id": ctx.member.id,
+            "month": month_option,
+            "day": day_option,
+            "real_or_un_birthday": real_or_un_birthday,
+            "last_event_datetime": datetime.datetime(year=2001, month=1, day=1),
+            "next_event_datetime": datetime.datetime(
+                year=2023, month=month_option, day=day_option, hour=16
+            ),
+            "created_datetime": datetime.datetime.now(tz=datetime.timezone.utc),
+        }
+        result = await mongo_motor_birthday_collection.insert_one(document)
+        logging.info("tried to insert mongo document")
+        await ctx.send("added.")
+
+    if DEV_MODE:
+        interval_seconds = 15
+    else:
+        interval_seconds = 4000  # a bit more than one hour
+
+    @Task.create(IntervalTrigger(seconds=interval_seconds))
+    async def create_birthday_events(self):
+        logging.info("In create_birthday_events")
+        server_birthday_event_opt_in_collection = self.bot.mongo_motor_db[
+            "server_birthday_event_opt_in_collection"
+        ]
+        mongo_motor_birthday_collection = self.bot.mongo_motor_db["birthdayCollection"]
+        for guild in self.bot.guilds:
+            # Define your search criteria
+            search_criteria = {"guild_id": guild.id}
+
+            # Find the latest document asynchronously
+            latest_documents = await server_birthday_event_opt_in_collection.find(
+                search_criteria
+            ).to_list(length=None)
+
+            # Sort the documents by date in descending order
+            latest_documents.sort(key=lambda x: x["created_date"], reverse=True)
+
+            # Check if there are any documents
+            if len(latest_documents) == 0:
+                continue
+
+            # Select first doc
+            opt_in_document = latest_documents[0]
+
+            if opt_in_document is None:
+                continue
+            if not opt_in_document["opt_in"]:
+                continue
+            seen_users = set()
+            async for birthday_document in mongo_motor_birthday_collection.find(
+                {"guild_id": {"$eq": guild.id}},
+                sort=[("created_datetime", pymongo.DESCENDING)],
+            ):
+                member_id = birthday_document["member_id"]
+                if member_id in seen_users:
+                    continue
+                seen_users.add(member_id)
+                event_date = datetime.datetime(
+                    year=2023,
+                    month=birthday_document["month"],
+                    day=birthday_document["day"],
+                )
+                now = datetime.datetime.now()
+                if (event_date - now).seconds > 0 and (event_date - now).days <= 3:
+                    if (
+                        (event_date - birthday_document["last_event_datetime"]).days
```

Comment:
does this account for leap year??? also im very confused on this portion actually. why did u check the seconds/days twice?

Reply:
This does not account for leap year. And due to hard coding elsewhere in the feature, it doesn't even handle next year.
Honestly this is ripe for a refactor. I'll add an issue about improving this section, but still merge this PR now.